File: Blog-Micro-services/Blog_ms2/src/main/java/projet/blog/Blog_ms2/entities/Article.java
package projet.blog.Blog_ms2.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;



@Data
@AllArgsConstructor
@NoArgsConstructor
public class Article  {

    private Long id;
    private String contenu;
    private String titre;
    private int claps;
    private String urlphoto;
    private Categorie categorie;
    Utilisateur auteur;
}

File: BlogApp/src/app/models/article.ts
import { User } from './auth';
import { Categorie } from './categorie';

export class Article {
  id: number | null;
  titre: string;
  contenu: string;
  claps: number;
  categorie: Categorie;
  auteur: number | User;

  constructor(
    id: number | null,
    titre: string,
    contenu: string,
    claps: number,
    categorie: Categorie,
    auteur: number | User
  ) {
    this.id = id;
    this.titre = titre;
    this.contenu = contenu;
    this.claps = claps;
    this.categorie = categorie;
    this.auteur = auteur;
  }
}
